server/pulp/server/db/model/__init__.py

```diff
@@ -768,6 +829,114 @@ class CeleryBeatLock(AutoRetryDocument):
     _ns = StringField(default='celery_beat_lock')
 
 
+class LazyCatalogEntry(AutoRetryDocument):
+    """
+    A catalog of content that can be downloaded by the specified plugin.
+
+    :ivar path: The content unit storage path.
+    :type path: str
+    :ivar importer_id: The ID of the plugin that contributed the catalog entry.
+        This plugin participates in the downloading of content when requested by the streamer.
+    :type importer_id: str
+    :ivar unit_id: The associated content unit ID.
+    :type unit_id: str
+    :ivar unit_type_id: The associated content unit type.
+    :type unit_type_id: str
+    :ivar url: The *real* download URL.
+    :type url: str
+    :ivar checksum: The checksum of the file associated with the
+        content unit. Used for validation.
+    :type checksum: str
+    :ivar checksum_algorithm: The algorithm used to generate the checksum.
+    :type checksum_algorithm: str
+    :ivar revision: The revision is used to group collections of entries.
+    :type revision: int
+    :ivar data: Arbitrary information stored with the entry.
+        Managed by the plugin.
+    :type data: dict
+    """
+
+    ALG_REGEX = r'(md5|sha1|sha224|sha256|sha384|sha512)'
+
+    meta = {
+        'collection': 'lazy_content_catalog',
+        'allow_inheritance': False,
+        'indexes': [
+            'importer_id',
+            {
+                'fields': [
+                    '-path',
+                    '-importer_id',
+                    '-revision',
+                ],
+                'unique': True
+            },
+        ],
+    }
+
+    # For backward compatibility
+    _ns = StringField(default=meta['collection'])
```

Comment:
We continue to model it because all models (created and not yet created) will have `_ns` added by PyMongo. This still occurs on master. PyMongo is set [here](https://github.com/pulp/pulp/blob/a828660f5fdaac279cd58ece43599cf412f6bcbd/server/pulp/server/db/connection.py#L150). We leave it enabled for consistency until we want to turn it off everywhere in a X release.